scripts/install-nix-from-closure.sh

```diff
@@ -1,181 +1,700 @@
 #!/bin/sh
+### v 2.02-1-dev
+###
+###############################
+###
+### Nix installation script
+### Universal shell script for for OS Linux environments
+###
+###############################
+###
+### Upstream URL:
+### https://github.com/NixOS/nix/blob/master/scripts/install-nix-from-closure.sh
+###
+### Script is fully POSIX compatible
+### (POSIX standard at http://pubs.opengroup.org/onlinepubs/9699919799/)
+### In some places sript can have controlled put of POSIX moments.
+###
+### Script strives to be fully transactional, as much as shell script can be.
+### That means, if script not succeeds - it does rolls back.
+###
+### Additional notes:
+### `/bin/sh -u` is not possible to do, because Docker environments has USER=''.
+###
+#
+###############################
+###  Documentation
+###############################
+{
+#
+#      Installer consist of:
+#      1. Setup environment
+#      2. Main constants
+#      3. CLI control constants
+#      4. CLI output functions
+#      5. Program stage functions
+#      6. Main function
+#
+#
+
+# Special things about this script
+#
+# 1) Script tries to be fully POSIX compatible, code is influenced by that.
+#
+# 2) Notice, Warning, Error, ErrorRevert level massages have forms to be
+# very readable and get attention, so they have special functions.
+# Message body starts from a new line.
+# And has 4 spaces from the left. Always.
+#
+###############################
+#
+#      notice "
+#
+#    Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users
+#    independently.
+#    "
+#
+###############################
+#
+# This is the best balance of code simplicity and code readability found so far.
+# This produces output in color:
+#
+###############################
+#
+#./install: Notice:
+#
+#    Install executed for ROOT.
+#
+#    Doing classic Linux package manager mode.
+#    In Nix this mode is called: single-user mode for root.
+#
+#    Nix can do multi-user mode, and manage package trees for users
+#    independently.
+#
+###############################
+#
+true # so {...} body has some code, shell will not permit otherwise
```

Comment:
You could also use `:`, but no need to replace it. Just if you wanted to save bandwidth.

Reply:
Many editors that support `sh` - allow to collapse `{..}` structures.
I don't sure that `:` is also supported.

Reply:
I remember even testing this in different IDEs.